src/Congo.WebApi/Data/Models/Product.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Congo.WebApi.Data.Models
+{
+    public class Product
+    {
+        [Key]
+        public string Id { get; init; } = Guid.NewGuid().ToString();
+
+        [Required]
```

Comment:
I'm not a big fan of `DataAnnotations` to be honest. If the desired result is to make this not nullable, I suggest we use `IEntityTypeConfiguration<T>` instead. Here is the doc page: https://docs.microsoft.com/en-us/ef/core/modeling/

It keeps the domain object pure and more POCO, and keeps the database-specific stuff away from it.

This goes for `[Key]` as well, of course.